clients/arweave_ipfs_s3.go

```diff
@@ -70,6 +62,10 @@ func DownloadDStorageFromGatewayList(u string, requestID string) (io.ReadCloser,
 	}
 
 	for _, gateway := range gateways {
+		if contains(gateway.String(), d.gatewaysAttempted) {
+			continue
+		}
+		d.gatewaysAttempted = append(d.gatewaysAttempted, gateway.String())
```

Comment:
Can you have a check against never trying any gateway because they're all exhausted? Say, on the first attempt we try all gateways and fail because there was a network hiccup. By the time we retry, the hiccup is gone but we've added all gateways to "attempted" here and don't try them anymore.

Could be either:
```go
if len(d.gatewaysAttempted) == gateways {
    d.gatewaysAttempted = []string{}
}
```

Or even, change the `gatewaysAttempted` to be a 'start index" instead, and every time we run this function we still try all the gateways, we just start from a different place and circle back with %. WDYT?

Reply:
Ahh yes that is much better have a 'start index', will update that